### PR TITLE
Cleanup vector matrix builder

### PR DESCRIPTION
**Pull request by endJunction**
Extend the SerialVectorMatrixBuilder such that all extra arguments are directly passed to the vector/matrix constructors. This allow for passing of options specific to the vector/matrix implementations w/o changing this interface.

Remove not any longer used SerialDenseSetup, SerialDenseVectorMatrixBuilder, and SerialLisVectorMatrixBuilder.
These configurations are handled by the NumericsConfig.h since a while and were used by two tests only; There an explicit instantiation is used now.

The SerialLinearSolver test is generalized and uses the NumericsConfig.h. (The LIS solver was not used before; now it is also checked if chosen in the numerics config.)

There are small reorderings of the code in NumericsConfig.h

### CODE REVIEW

File: Tests/AssemblerLib/TestSerialVectorMatrixBuilder.cpp [hunk 1]
```diff
@@ -80,8 +76,8 @@ TYPED_TEST_P(AssemblerLibSerialVectorMatrixBuilder, createVector)
 TYPED_TEST_P(AssemblerLibSerialVectorMatrixBuilder, createMatrix)
 {
     typedef typename TestFixture::MatrixType M;
-    typedef TypeParam SerialBuilder;
-    M* v = SerialBuilder::createMatrix(this->cmap->size());
+    typedef TypeParam Builder;
+    M* v = Builder::createMatrix(this->cmap->size());
```
Comment (TomFischer) [inline Tests/AssemblerLib/TestSerialVectorMatrixBuilder.cpp]: Why not ```M *m ...``` instead of ```M *v ...```?

**Comment by TomFischer** [comment]
:+1:

**Comment by norihiro-w** [comment]
:+1:

**Comment by bilke** [comment]
@endJunction The [Windows build on Jenkins](https://svn.ufz.de:8443/job/OGS-6/job/Win/135/console) errors:

 f:\jenkins\workspace\OGS-6\Win\ogs\Tests\AssemblerLib\TestSerialLinearSolver.cpp(128): error C2899: typename cannot be used outside a template declaration [F:\jenkins\workspace\OGS-6\Win\build\Tests\testrunner.vcxproj]

Can you please have a look?

**Comment by endJunction** [comment]
@bilke WIP at [endJunction/master](https://github.com/endJunction/ogs/commit/1143837a75d5dcc5c443a64dc48d8d54adb178ce).

**Pull request merged**